### PR TITLE
lead regoin role command bug fix

### PR DESCRIPTION
**Pull request by bowenjw**
Missed reference to states in lead region role command added the following helper functions to state.ts

```ts
function isMemberStateLead(member: GuildMember): Promise<boolean>

function isStateLeadRole(role: Role): boolean

function memberStates(member: GuildMember): Collection<string, Role>
```

### CODE REVIEW

File: src/structures/states.ts [hunk 1]
```diff
@@ -123,3 +127,18 @@ export type StateAbbreviation =
 
 export const states = new Collection<StateAbbreviation, state>();
 statesConfig.map((s) => states.set(s.abbreviation.toLocaleLowerCase() as StateAbbreviation, s));
+
+export async function isMemberStateLead(member: GuildMember) {
+	const role = await member.guild.roles.fetch(stateLeadRoleID);
```
Comment (Lewdcario) [inline src/structures/states.ts:132]: No need to use a promise here - roles should be cached

File: src/structures/sme.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+export function getSMERoles(member: GuildMember) {
+	if (!SMERoleIDs) throw new Error('SME_ROLE_IDS not present in .env');
+
+	const SMERoles = new Collection<Snowflake, Role>();
+
+	SMERoleIDs.forEach((id) => {
+		if (member.roles.cache.has(id)) SMERoles.set(id, member.roles.cache.get(id));
+	});
+
+	return SMERoles;
```
Comment (Lewdcario) [inline src/structures/sme.ts:16]: I would prefer a smaller version of this:
```ts
return member.roles.filter(id => SMERoleIDs.includes(id));
```

File: src/structures/states.ts [hunk 3]
```diff
@@ -123,3 +127,18 @@ export type StateAbbreviation =
 
 export const states = new Collection<StateAbbreviation, state>();
 statesConfig.map((s) => states.set(s.abbreviation.toLocaleLowerCase() as StateAbbreviation, s));
+
+export function isMemberStateLead(member: GuildMember) {
+	const role = member.guild.roles.cache.get(stateLeadRoleID);
+	if (!role) throw Error('Missing Valid STATE_LEAD_ROLE_ID in .env');
```
Comment (Lewdcario) [inline src/structures/states.ts:133]: Misleading error, we should check for the env being present first, and then return false if the role isn't present in their list of roles

**Review by Lewdcario — CHANGES_REQUESTED** [review]

File: src/structures/states.ts (hunk 3 above)
Comment (bowenjw) [reply src/structures/states.ts:133]: 132 gets the role from the guild
so it check both if the env is present and if the role is in the guild

File: src/structures/sme.ts (hunk 2 above)
Comment (bowenjw) [reply src/structures/sme.ts:16]: sorry this was not meant to included this change in the this fix some of my code leaked over on accident. I will make the change

File: src/structures/states.ts [hunk 4]
```diff
@@ -123,3 +127,19 @@ export type StateAbbreviation =
 
 export const states = new Collection<StateAbbreviation, state>();
 statesConfig.map((s) => states.set(s.abbreviation.toLocaleLowerCase() as StateAbbreviation, s));
+
+export function isMemberStateLead(member: GuildMember) {
+	if (!stateLeadRoleID) throw Error('Missing STATE_LEAD_ROLE_ID in .env');
+	const role = member.guild.roles.cache.get(stateLeadRoleID);
+	if (!role) throw Error('inalid Role ID please check STATE_LEAD_ROLE_ID in .env');
```
Comment (Lewdcario) [inline src/structures/states.ts:134]: Invalid

File: src/commands/chat/execution/lead/region/role.ts [hunk 5]
```diff
@@ -20,7 +20,7 @@ export default async function execute(interaction: ChatInputCommandInteraction<'
 	const stateLead = interaction.member;
 
 	// If the state lead and the target member do not have the same state role
-	if (!memberState(stateLead).some((role) => memberState(target).has(role.id))) {
+	if (!memberStates(stateLead).some((role) => memberStates(target).has(role.id))) {
```
Comment (Lewdcario) [inline src/commands/chat/execution/lead/region/role.ts:23]: image

File: src/commands/chat/execution/lead/region/role.ts [hunk 6]
```diff
@@ -48,7 +48,7 @@ export default async function execute(interaction: ChatInputCommandInteraction<'
 		key: 'role-success-remove',
 		locale,
 		ns,
-		args: { tag: target.user.tag }
+		args: { role: regionLeadRole.toString(), tag: target.user.tag }
```
Comment (Lewdcario) [inline src/commands/chat/execution/lead/region/role.ts:51]: image

**Review by Lewdcario — APPROVED** [review]

**Pull request merged**